packages/rest/src/router/metadata.ts

```diff
@@ -304,6 +370,28 @@ export function param(paramSpec: ParameterObject) {
             ' is not allowed.',
         );
       }
+      // Map design-time parameter type to the OpenAPI param type
+      if (paramSpec && !paramSpec.type) {
```

Comment:
The location of this code makes it so that it only applies to parameter decorators and not method decorators. Is this intentional?